src/main/java/org/apache/sling/event/impl/jobs/config/JobManagerConfiguration.java

```diff
@@ -308,21 +306,21 @@ public boolean isActive() {
      * This ResourceResolver provides read and write access to all resources relevant for the event
      * and job handling.
      * 
-     * @return A resource resolver or {@code null} if the component is already deactivated.
+     * @return A resource resolver 
      * @throws RuntimeException if the resolver can't be created.
      */
-    public ResourceResolver createResourceResolver() {
-        ResourceResolver resolver = null;
+    public @NotNull ResourceResolver createResourceResolver() {
```

Comment:
IIUC this change will result in exceptions rather than silent swallows if the factory is null. That might its own consequences. What is the reasoning behind this change?